lib/logstash/filters/date.rb

```diff
@@ -202,6 +214,24 @@ def filter(event)
           last_exception = RuntimeError.new "Unknown"
           fieldparsers.each do |parserconfig|
             parserconfig[:parser].each do |parser|
+              if @iso8601_format
```

Comment:
We should do high-precision support for any format (ISO8601, or otherwise) that has the "S" in the format string (which is subsecond)